### DISCUSSION TITLE
What to build in 2023?

### DISCUSSION THREAD

**Discussion by ivarconr** [Unleash/unleash#2878] [Ideas]
What features would you like to see in Unleash in 2023?

**Comment by ivarconr** [comment]
One common request from request from Unleash Admins is to introduce a **maintenance mode**, where you can lock down updates in Unleash during important events (Black Friday, database migrations etc). We are building that and will ship the beta in next version in Unleash.

https://github.com/Unleash/unleash/issues/2699

**Comment by ivarconr** [comment]
We have updated the [public Unleash Roadmap for Q1-2023](https://github.com/orgs/Unleash/projects/10). 

If you see an enhancement that is important for you, simply do a 👍🏼 on the issue request to show your interest. This will help us prioritize.

**Comment by krugi** [comment]
I'd love to have Slack events for health related stuff, like when Unleash marks a flag as Potentially Stale
<img width="606" alt="Screenshot 2023-01-31 at 16 16 09" src="https://user-images.githubusercontent.com/15140079/215787084-3258f76c-5daf-4f13-aa97-2ac2710418a9.png">

**Reply by ivarconr** [reply]
This is an excellent suggestion. We are exploring a more thought through [notification system in Unleash](https://github.com/Unleash/unleash/issues/2945), and think this is somewhat related.

**Comment by singhprincejeet** [comment]
A nice feature to have would be the ability to assign multiple roles to a user/group.

The benefit of this would be the ability to have more specific roles which can then be assigned or unassigned easily to a user/group.

This would be similar to GCP IAM where a user/group can have a Storage Admin and Kubernetes Editor role at same time.

**Reply by ivarconr** [reply]
We have started working on this. If you reach out to the team on slack we might be able to put you in the early beta group.

**Comment by albertodearriba** [comment]
I know this is a big feature but Dynamic Config inside Unleash would be great. I mean the option of storing other values beyond booleans (strings, mainly)

**Reply by ivarconr** [reply]
Have you considered to use [variant payloads](https://docs.getunleash.io/reference/feature-toggle-variants#variant-payload)? We will soon also allow you to configure variants differently per environment (#2254)

**Reply by albertodearriba** [reply]
I know you can store other values in variants but that workaround would work for small features and it would be messy to handle many config variables that way

**Reply by ivarconr** [reply]
Got it. We have carefully not built configuration yet, as it is easy to miss-use it for static application configuration instead of more dynamic feature configuration. I have capture this suggestion on our idea-board. Thanks!

**Comment by HarshCodes-07** [comment]
SDK for NUXT SSR

**Reply by ivarconr** [reply]
Anyone who want to help build a MVP for this? We did a similar thing for Next.js recently ([source](https://github.com/Unleash/unleash-client-nextjs)) and we can probably do similar things for Nuxt. It would be awesome to get some help from people more familiar with the Nuxt Ecosystem to start the initiative, and we can help on the Unleash side of things!

**Comment by jlane-tkww** [comment]
It would be helpful to have a last seen timestamp per environment. We have a use case where product managers would like to know when they toggle a flag to be enabled in prod, they’d like to see that timestamp for that environment specifically once it has been called for the first time. However, as is, it’s hard to know which environment the last seen timestamp is referring to. It would also be nice if the GET call to see details on a single feature would list the last seen for each environment as well.  I know we can get this as well as more detailed data from metrics, but it would be nice if this small detail was included in the environment section of the response as well. Thanks for the consideration!

**Reply by ivarconr** [reply]
Hi @jlane-tkww,

we have built an initial solution for this, you can check it our on the demo instance ([link](https://app.unleash-hosted.com/demo/projects/demo-app?)). Let us know what you think!
![image](https://github.com/Unleash/unleash/assets/158948/d1a85425-0e13-4248-8505-a4f17c3bb43d)

**Reply by jlane-tkww** [reply]
@ivarconr - that looks great and will be super helpful!! Is this available via the api as well? Thanks for working on this!

**Reply by gardleopard** [reply]
If you look at https://docs.getunleash.io/reference/api/unleash/get-features you can see that its present in the featureEnvironmentSchema.

**Comment by sjaanus** [comment]
Closing since 2023 is over.
